### PR TITLE
Adding scm webhook api url as part of  webhook creation

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### CODE REVIEW

File: jazz-build-module/scm-module.groovy [hunk 1]
```diff
@@ -164,13 +164,16 @@ def setRepoPermissions(repo_owner, repo_name, admin_group) {
 }
 
 def addWebhook(repo_name, webhookName,scm_webhook_target_url) {
+    def scm_events_webhook_url = "https://${scm_config.API.PROD['*']}/api/jazz/scm-web-hook"
```
Comment (suryajak) [inline jazz-build-module/scm-module.groovy]: this line of code should be in service onboarding. Moreover use utils to grab the API Id (as we have done in other places).

**Comment by suryajak** [comment]
We also need to send events for service onboarding so that environment handler can pick them up.

File: jazz-build-module/scm-module.groovy (hunk 1 above)
Comment (SiniWilson) [reply jazz-build-module/scm-module.groovy]: Fixed

File: service-onboarding-build-pack/Jenkinsfile [hunk 2]
```diff
@@ -221,44 +223,50 @@ node  {
...
             withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: repo_credential_id, passwordVariable: 'PWD', usernameVariable: 'UNAME']]) {
 				try {
+					events.sendStartedEvent("ADD_WEBHOOK", 'service repo creation starts', context_map)
 					def scm_webhook_target_url
 					if (configLoader.SCM.TYPE == "gitlab") {
 						scm_webhook_target_url = "http://${configLoader.JENKINS.USERNAME}:${configLoader.JENKINS.JENKINS_PASSWORD}@${JenkinsLocationConfiguration.get().getUrl().split('/')[2]}/project/Gitlab-Trigger-Job"
 					} else if (configLoader.SCM.TYPE == "bitbucket") {
 						scm_webhook_target_url = JenkinsLocationConfiguration.get().getUrl() + "/bitbucket-scmsource-hook/notify"
 					}
+					scmModule.setScmEventsWebhookUrl("${g_base_url}/jazz/scm-web-hook")
```
Comment (suryajak) [inline service-onboarding-build-pack/Jenkinsfile]: Why aren't we calling the `scmModule.addWebhook` with `"${g_base_url}/jazz/scm-web-hook"` as the argument?

File: jazz-build-module/scm-module.groovy [hunk 3]
```diff
@@ -163,74 +164,84 @@ def setRepoPermissions(repo_owner, repo_name, admin_group) {
...
-		def scm_webhook_api = "${scm_protocol}${scm_config.REPOSITORY.BASE_URL}/api/v4/projects/${proj_id}/hooks?enable_ssl_verification=false&push_events=true&url="
-		sh "curl --header \"Private-Token: ${scm_config.SCM.PRIVATE_TOKEN}\" -X POST \"${scm_webhook_api}$scm_webhook_target_url\""
-    }else if(scm_config.SCM.TYPE == "bitbucket"){
-		def scm_webhook_api = "${scm_protocol}${scm_config.REPOSITORY.BASE_URL}/rest/webhook/1.0/projects/${scm_config.REPOSITORY.REPO_BASE_SERVICES}/repos/"
-		sh "curl -X PUT -k -v -u \"${scm_config.SCM.USERNAME}:${scm_config.SCM.PASSWORD}\" -H \"Content-Type: application/json\" ${scm_webhook_api}${repo_name}/configurations  -d \'{\"title\": \"${webhookName}\", \"url\": \"${scm_webhook_target_url}\" , \"enabled\": true}\'"
+        def scm_webhook_api = "${scm_protocol}${scm_config.REPOSITORY.BASE_URL}/api/v4/projects/${proj_id}/hooks?enable_ssl_verification=false&push_events=true&url="
+        sh "curl --header \"Private-Token: ${scm_config.SCM.PRIVATE_TOKEN}\" -X POST \"${scm_webhook_api}$scm_webhook_target_url\""
+        sh "curl --header \"Private-Token: ${scm_config.SCM.PRIVATE_TOKEN}\" -X POST \"${scm_webhook_api}$scm_events_webhook_url\""
```
Comment (suryajak) [inline jazz-build-module/scm-module.groovy]: This doesn't look right where now the consumer of this method doesn't know that we add two webhooks. Let me the consumer twice for each webhook url they want to add. 

In short remove this " sh "curl --header \"Private-Token: ${scm_config.SCM.PRIVATE_TOKEN}\" -X POST \"${scm_webhook_api}$scm_events_webhook_url\"""

File: jazz-build-module/scm-module.groovy [hunk 4]
```diff
@@ -163,74 +164,84 @@ def setRepoPermissions(repo_owner, repo_name, admin_group) {
...
-    }else if(scm_config.SCM.TYPE == "bitbucket"){
+    } else if (scm_config.SCM.TYPE == "bitbucket") {
         return "${scm_protocol}${scm_config.REPOSITORY.BASE_URL}/scm/${scm_config.REPOSITORY.REPO_BASE_PLATFORM}/${repo_name}.git"
     }
 }
 
+/**
+ * Set webhook url
+ * @return      
+ */
```
Comment (suryajak) [inline jazz-build-module/scm-module.groovy]: remove this method (look at the comment above for why)

File: service-onboarding-build-pack/Jenkinsfile (hunk 2 above)
Comment (SiniWilson) [reply service-onboarding-build-pack/Jenkinsfile]: Fixed

File: jazz-build-module/scm-module.groovy (hunk 3 above)
Comment (SiniWilson) [reply jazz-build-module/scm-module.groovy]: Fixed

File: jazz-build-module/scm-module.groovy (hunk 4 above)
Comment (SiniWilson) [reply jazz-build-module/scm-module.groovy]: Fixed

**Pull request merged**